### ISSUE TITLE
Attempt to invoke interface method 'io.flutter.plugin.common.BinaryMessenger io.flutter.plugin.common.PluginRegistry$Registrar.messenger

### ISSUE THREAD

**Issue by sky126** [pichillilorenzo/flutter_inappwebview#220]
I create new project  only add  flutter_inappwebview  

Flutter version:  Flutter (Channel beta, v1.12.13+hotfix.6, on Microsoft Windows [Version 10.0.18363.476], locale zh-CN) 

Plugin version:  ^2.1.0+1 

## Stacktrace/Logcat 
2019-12-12 20:11:53.794 9623-9623/com.example.diantui E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.diantui, PID: 9623
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.example.diantui/com.example.diantui.MainActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'io.flutter.plugin.common.BinaryMessenger io.flutter.plugin.common.PluginRegistry$Registrar.messenger()' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2913)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'io.flutter.plugin.common.BinaryMessenger io.flutter.plugin.common.PluginRegistry$Registrar.messenger()' on a null object reference
        at com.pichillilorenzo.flutter_inappwebview.InAppBrowser.<init>(InAppBrowser.java:66)
        at com.pichillilorenzo.flutter_inappwebview.InAppWebViewFlutterPlugin.onAttachedToEngine(InAppWebViewFlutterPlugin.java:40)
        at io.flutter.embedding.engine.FlutterEnginePluginRegistry.add(FlutterEnginePluginRegistry.java:125)
        at io.flutter.plugins.GeneratedPluginRegistrant.registerWith(GeneratedPluginRegistrant.java:15)
        at com.example.diantui.MainActivity.configureFlutterEngine(MainActivity.kt:11)
        at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onAttach(FlutterActivityAndFragmentDelegate.java:185)
        at io.flutter.embedding.android.FlutterActivity.onCreate(FlutterActivity.java:418)
        at android.app.Activity.performCreate(Activity.java:7136)
        at android.app.Activity.performCreate(Activity.java:7127)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2893)

**Comment by sky126** [comment]
ShimPluginRegistry shimPluginRegistry = new ShimPluginRegistry(binding.getFlutterEngine());
    registrar = shimPluginRegistry.registrarFor("com.pichillilorenzo/flutter_inappwebview");
    inAppBrowser = new InAppBrowser(registrar);

**Comment by tatsuyuki25** [comment]
same issue

**Comment by pichillilorenzo** [comment]
I'm working on it to use the new Java API for Flutter embedding v2

**Comment by fulus06** [comment]
同

**Comment by appstute-pratik** [comment]
Same issue

**Comment by asdfgh5889** [comment]
> Same issue, Any work around?

Getting dependency directly from git solved the problem. I think in couple days this fix will be released.

```
  flutter_inappwebview:
    git:
      url: https://github.com/pichillilorenzo/flutter_inappwebview.git
      ref: master
```

**Comment by MostafaEisam** [comment]
Same issue

**Comment by HakuFlou** [comment]
+1

**Comment by appstute-pratik** [comment]
` flutter_inappwebview:
    git:
      url: https://github.com/pichillilorenzo/flutter_inappwebview.git
      ref: master`

Is this fix safe for production build?

**Comment by appstute-pratik** [comment]
> > Same issue, Any work around?
> 
> Getting dependency directly from git solved the problem. I think in couple days this fix will be released.
> 
> ```
>   flutter_inappwebview:
>     git:
>       url: https://github.com/pichillilorenzo/flutter_inappwebview.git
>       ref: master
> ```

Is this fix safe for production build?

**Comment by asdfgh5889** [comment]
> > > Same issue, Any work around?
> > 
> > 
> > Getting dependency directly from git solved the problem. I think in couple days this fix will be released.
> > ```
> >   flutter_inappwebview:
> >     git:
> >       url: https://github.com/pichillilorenzo/flutter_inappwebview.git
> >       ref: master
> > ```
> 
> Is this fix safe for production build?

I'm not completely sure, but should be ok

**Comment by appstute-pratik** [comment]
I'm getting java.lang.UnsatisfiedLinkError: dalvik.system.PathClassLoader error for play store version.

**Comment by appstute-pratik** [comment]
> I'm working on it to use the new Java API for Flutter embedding v2

Hi, Any workaround? I'm getting java.lang.UnsatisfiedLinkError error for a production build on play store.

Following code added in pubspec.
`flutter_inappwebview:
    git:
      url: https://github.com/pichillilorenzo/flutter_inappwebview.git
      ref: master`

Thanks in advance.

**Comment by sidpram** [comment]
> > > > Same issue, Any work around?
> > > 
> > > 
> > > Getting dependency directly from git solved the problem. I think in couple days this fix will be released.
> > > ```
> > >   flutter_inappwebview:
> > >     git:
> > >       url: https://github.com/pichillilorenzo/flutter_inappwebview.git
> > >       ref: master
> > > ```
> > 
> > 
> > Is this fix safe for production build?
> 
> I'm not completely sure, but should be ok

If we include flutter_inappwebview as
 flutter_inappwebview:
    git:
      url: https://github.com/pichillilorenzo/flutter_inappwebview.git
      ref: master

Then it gives error on onPermissionRequest and initialOptions.  Unable to resolve these two functions

**Comment by espresso3389** [comment]
I have a plugin depending on the module and pub.dev does not allow me to release a plugin that depends on github; the workaround does not work for me :(
Any ETA for a new release?

> Getting dependency directly from git solved the problem. I think in couple days this fix will be released.
> 
> ```
>   flutter_inappwebview:
>     git:
>       url: https://github.com/pichillilorenzo/flutter_inappwebview.git
>       ref: master
> ```

**Comment by lehno** [comment]
Any fix on it, cant get this to work

**Comment by soulduse** [comment]
Still not working..

**Comment by PhanHug93** [comment]
In InAppWebViewFlutterPlugin.class i fixed: 
```
 public InAppWebViewFlutterPlugin(PluginRegistry.Registrar registrar) {
        this.registrar = registrar;
    }
```
In GeneratedPluginRegistrant class i fixed: 
    `flutterEngine.getPlugins().add(new com.pichillilorenzo.flutter_inappwebview.InAppWebViewFlutterPlugin(shimPluginRegistry.registrarFor("com.pichillilorenzo/flutter_inappwebview")));`
 Code work but it's only local code ..

**Comment by alexlovar** [comment]
> > Same issue, Any work around?
> 
> Getting dependency directly from git solved the problem. I think in couple days this fix will be released.
> 
> ```
>   flutter_inappwebview:
>     git:
>       url: https://github.com/pichillilorenzo/flutter_inappwebview.git
>       ref: master
> ```

same error here, this works for me. thanks

**Comment by pichillilorenzo** [comment]
It has been fixed with version `3.0.0`+.
@alexlovar you shouldn't use the master branch anymore.
